spec/requests/topics_controller_spec.rb

```diff
@@ -1011,6 +1011,46 @@ def topic_user_post_timings_count(user, topic)
           end
         end
 
+        context 'updating to a category with restricted tags' do
+          let!(:category) { Fabricate(:category) }
+          let!(:restricted_category) { Fabricate(:category) }
+          let!(:tag1) { Fabricate(:tag, name: 'tag1') }
+          let!(:tag2) { Fabricate(:tag, name: 'tag2') }
+
+          before do
+            SiteSetting.tagging_enabled = true
+            topic.update!(tags: [tag1])
+          end
+
+          it 'can change to a category disallowing this topic current tags' do
+            restricted_category.allowed_tags = [tag2.name]
+            restricted_category.reload
+
+            put "/t/#{topic.slug}/#{topic.id}.json", params: { category_id: restricted_category.id }
+
+            result = ::JSON.parse(response.body)
+
+            expect(response.status).to eq(422)
+            expect(result['errors']).to eq([I18n.t('category.errors.disallowed_topic_tags', tags: [tag1].map(&:name).join(", "))])
+            expect(topic.reload.category_id).not_to eq(restricted_category.id)
+          end
+
+          it 'can change to a category allowing this topic current tags' do
+            restricted_category.allowed_tags = [tag1.name]
+            restricted_category.reload
```

Comment:
These `restricted_category.reload` calls are confusing to me. Why would this be necessary unless your test is checking the reloaded value?

Reply:
Yes for some reason I thought it was needed, but no. Thanks.